### PR TITLE
feat: support arm64 compilation

### PR DESCRIPTION
**Pull request by guillaumeloftorbital**
**Description**
- Use qemu to build for arm64 cpp and python package 
related to https://github.com/open-space-collective/open-space-toolkit/pull/96

### CODE REVIEW

File: docker/development/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,9 @@
 # Apache License 2.0
 
-ARG BASE_IMAGE_VERSION="latest"
+ARG BASE_IMAGE_VERSION="0.5.8-12-g1840e81"
```
Comment (guillaumeloftorbital) [inline docker/development/Dockerfile:3]: To be updated when the  change on the base

**Review by apaletta3 — APPROVED** [review]
Looks great, thanks!

**Review by vishwa2710 — APPROVED** [review]
Looks good after you address Antoine's comments!

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -44,7 +44,7 @@ jobs:
     name: Build Development Image
     needs:
       - prepare-environment
-    uses: open-space-collective/open-space-toolkit/.github/workflows/build-image.yml@main
+    uses: open-space-collective/open-space-toolkit/.github/workflows/build-image.yml@users/guillaume/multi-arch
```
Comment (Derollez) [inline .github/workflows/build-test.yml:47]: switch back to `main` when merged on base repo and adapt twice below as well 🙏

File: CMakeLists.txt [hunk 3]
```diff
@@ -224,8 +224,20 @@ SET (Boost_USE_MULTITHREADED ON)
...
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
+    # Example path adjustment for aarch64, adjust the path as needed
+    set(BACKTRACE_INCLUDE "/usr/lib/gcc/aarch64-linux-gnu/9/include/backtrace.h")
+else()
+    message(FATAL_ERROR "Unsupported architecture: ${CMAKE_SYSTEM_PROCESSOR}")
+endif()
+
+# Add definitions with architecture-specific path
+add_definitions(-DBOOST_STACKTRACE_USE_BACKTRACE)
+add_definitions(-DBOOST_STACKTRACE_BACKTRACE_INCLUDE_FILE=<${BACKTRACE_INCLUDE}>)
```
Comment (Derollez) [inline CMakeLists.txt:240]: I don't know what is the standard for MakeFile instructions between lower and upper caps but let's adapt here to follow the upper caps scheme in our current file? 🙏

Comment (Derollez) [inline CMakeLists.txt:240]:
```suggestion

# Detect system architecture

IF (CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64")

    # Architecture-specific include for x86_64
    SET (BACKTRACE_INCLUDE "/usr/lib/gcc/x86_64-linux-gnu/9/include/backtrace.h")
    
ELSEIF (CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")

    # Example path adjustment for aarch64, adjust the path as needed
    SET (BACKTRACE_INCLUDE "/usr/lib/gcc/aarch64-linux-gnu/9/include/backtrace.h")
  
ELSE ()

    MESSAGE (FATAL_ERROR "Unsupported architecture: ${CMAKE_SYSTEM_PROCESSOR}")
    
ENDIF ()

# Add definitions with architecture-specific path

ADD_DEFINITIONS (-DBOOST_STACKTRACE_USE_BACKTRACE)
ADD_DEFINITIONS (-DBOOST_STACKTRACE_BACKTRACE_INCLUDE_FILE=<${BACKTRACE_INCLUDE}>)
```

Comment (Derollez) [reply CMakeLists.txt:240]: (suggestion below)

File: bindings/python/CMakeLists.txt [hunk 4]
```diff
@@ -112,6 +112,15 @@ FUNCTION (PY_ADD_PACKAGE_DIRECTORY NAME)
...
     SET (PACKAGE_DIRECTORY_NAME ${NAME})
 
+    # Set platforme name for wheel
+    if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64")
+        SET (PLATFORM "linux_x86_64")
+    ELSEIF(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
+        SET (PLATFORM "linux_aarch64")
+    ELSE()
+        MESSAGE (FATAL_ERROR "Unsupported platform")
+    ENDIF()
```
Comment (Derollez) [inline bindings/python/CMakeLists.txt:122]:
```suggestion
    # Set platform name for wheel
    IF (CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64")
        SET (PLATFORM "linux_x86_64")
    ELSEIF (CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
        SET (PLATFORM "linux_aarch64")
    ELSE ()
        MESSAGE (FATAL_ERROR "Unsupported platform")
    ENDIF ()
```

**Review by Derollez — COMMENTED** [review]
That's awesome, thanks a lot @guillaumeloftorbital for adding those changes 🙏

**Pull request merged**